opensquirrel/default_gate_modifiers.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+from collections.abc import Callable
+from typing import Any, SupportsFloat
+
+from opensquirrel.ir import BlochSphereRotation, ControlledGate, QubitLike
+
+
+class GateModifier:
+    pass
+
+
+class InverseGateModifier(GateModifier):
+    def __init__(self, generator_f_gate: Callable[..., BlochSphereRotation]) -> None:
+        self.generator_f_gate = generator_f_gate
+
+    def __call__(self, *args: Any) -> BlochSphereRotation:
+        gate: BlochSphereRotation = self.generator_f_gate(*args)
+        modified_angle = gate.angle * -1
+        return BlochSphereRotation(qubit=gate.qubit, axis=gate.axis, angle=modified_angle, phase=gate.phase)
+
+
+class PowerGateModifier(GateModifier):
```

Comment:
Can a power exponent be imaginary? If so, `float` should be replaced with `np.complex128`

Reply:
The power cannot be imaginary, it is either an integer or a float.